pyk/src/pyk/cterm.py

```diff
@@ -16,10 +16,15 @@ class CTerm:
 
     def __init__(self, term: KInner) -> None:
         config, constraint = splitConfigAndConstraints(term)
-        constraints = tuple(flattenLabel('#And', constraint))
+        constraints = tuple(sorted(set(flattenLabel('#And', constraint)), key=self._constraint_sort_key))
         object.__setattr__(self, 'config', config)
         object.__setattr__(self, 'constraints', constraints)
 
+    @staticmethod
+    def _constraint_sort_key(term: KInner) -> Tuple[int, str]:
+        term_str = str(term)
+        return (len(term_str), term_str)
+
```

Comment:
I think let's add a `_normalize_constraints` function which both does this sorting and other trivial simplifications like this one: https://github.com/runtimeverification/erc20-verification/pull/428/files#r872984309

Reply:
And call it whenever the term or constraint changes (and at initialization of course).

Reply:
Good idea.

> call it whenever the term or constraint changes
Thankfully there's no need for that, this class is (designed to be) immutable.

Reply:
Nice :)

Reply:
Done.